services/subnet_manager/src/main/java/com/futurewei/alcor/subnet/service/SubnetService.java

```diff
@@ -32,7 +35,13 @@ public void fallbackOperation (AtomicReference<RouteWebJson> routeResponseAtomic
     // Allocate Gateway Mac
     public MacStateJson allocateMacGateway (String projectId, String vpcId, String portId) throws Exception;
 
-    // TODO : Verify/Allocate Gateway IP, subnet id, port id, subnet cidr, response:IP - unique
-    public IpAddrRequest allocateIPGateway (String subnetId) throws Exception;
+    // Verify/Allocate Gateway IP
+    public IpAddrRequest allocateIPGateway (String subnetId, String cidr) throws Exception;
```

Comment:
To be accurate, let us use "allocateIpAddressForGatewayPort"

Reply:
Also inside this method, I believe we have two standalone logics. One is to allocate Ip range for subnet, and the other is to allocate ip address for gateway port. Let us split them into two methods.